build.gradle

```diff
@@ -252,7 +252,7 @@ subprojects {
       // code in the test (best guess is that it is tail output from last test). We won't have
       // an output file for these, so simply ignore them. If they become critical for debugging,
       // they can be seen with showStandardStreams.
-      if (td.name == td.className) {
```

Comment:
The problem with just ignoring here is that we don't really have an explanation for how this can happen. I forget the exact details of what the output was (other than note that it might be "leftover" from buggy test runner, and my immediate thought that some could also be from things like static initialization). If anything, we actually might want to log *more info. Aside from potential log size (and noise) I don't remember why I didn't include the `toe.message` in the exception handler logging below.

Have we run with that modification to see if it's possible to tell what type of log output is getting lost? We could also change `ERROR` to `WARNING`. The main reason I wanted it at the scarier `ERROR` level was because it was unclear at the time just how frequent it would be, and so there was some indication in the log that some output from *some* test that was running before that one might have missing log data.

We also might not need the stack trace, which adds more noise. I think I included that so if we saw it consistently we could try to track down when this would commonly occur (e.g. if we could track it down to something like static initialization as the common cause).

Reply:
Hi @ewencp , thanks for the review!

It wasn't clear to me that we think it shouldn't happen. With that knowledge, it does seem worthwhile to include some debugging logs to try and understand what is going on.

FWIW, I like the stacktrace. That's actually the thing that tipped me off that the problem was with the build itself, rather than a test, thereby saving me from hours of digging in the wrong direction.

I might be an idealist, but I'm not worried about the "level of noise", as in "number of log lines". The fact is that the build system isn't behaving the way we think it should, so we should try to figure out why. Once we understand why, we'll know what to do, and _then_ we won't have to listen to any more noise.

I'll update the PR.

Reply:
Yeah, unfortunately iirc it previously looked like Gradle's implementation of this feature was just buggy so might not be completely fixable without figuring out what to fix upstream. I prefer more info as well, but since people already often have trouble finding the real source of issues in long logs, I am a bit hesitant to make them even more voluminous. I think current modification is reasonable tradeoff.